sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/RetryPolicy.java

```diff
@@ -59,11 +86,25 @@ public RetryPolicy() {
      * @throws NullPointerException When {@code retryAfterTimeUnit} is null and {@code retryAfterHeader} is not null.
      */
     public RetryPolicy(String retryAfterHeader, ChronoUnit retryAfterTimeUnit) {
-        this(new ExponentialBackoff(), retryAfterHeader, retryAfterTimeUnit);
+        this.retryAfterHeader = HeaderName.fromString(retryAfterHeader);
```

Comment:
I wonder if, instead of keeping this constructor we move this into a function for RetryStrategy, and something RetryOptions could configure. This was used for a very limited case initially for AppConfiguration and I believe `ImplUtils.getRetryAfterFromHeaders` handles that. And an option in RetryOptions like `Function<HttpHeaders, Duration>` seems to be a better option anyway.

Reply:
Yeah, I think this could be a good addition later on to RetryOptions and can be introduced when needed. For now I will remove this constructor from RetryPolicy.